srml/staking/src/lib.rs

```diff
@@ -16,7 +16,157 @@
+//! - Controller account.
+//! - Stash account, not increasing the staked value.
+//! - Stash account, also increasing the staked value.
+//! 
+//! ### Slashing details 
+//! 
+//! - A validator can be _reported_ to be offline at any point via `on_offline_validator` public function. 
+//! - Each validator declares how many times they can be _reported_ before it actually gets slashed via the `unstake_threshold` in `ValidatorPrefs`. On top of this, the module also introduces a `OfflineSlashGrace`, which applies to all validators and prevents them from getting immediately slashed.
+//! - Similar to the reward value, the slash value is updated at the end of each era by multiplying `slot_stake` and a configuration storage item, `OfflineSlash`.
+//! - Once a validator has been reported a sufficient amount of times, the actual value that gets deducted from that validator, and every single nominator that voted for it is calculated by multiplying the result of the above point by `2.pow(unstake_threshold)`.
+//! - If the previous overflow, then `slot_stake` is used.
+//! - If the previous is more than what the validator/nominator has in stake, all of their stake is slashed (`.max(total_stake)` in other words).
+//! 
+//! ### Additional Fund Management Operations
+//! 
+//! Any funds already placed into stash can be the target of the following operations:
+//! 
+//! - The controller account can free an portion (or all) of the funds using the `unbond()` call. Note that the funds are not immediately accessible, instead, a duration denoted by `BondingDuration` number of eras must pass until the funds can be actually removed.
+//! - To actually remove the funds, once the bonding duration is over, the `withdraw_unbonded()` can be used.
+//! - As opposed to the above, additional funds can be added to the stash account via the `bond_extra()` transaction call. 
+//! 
+//! ### Election algorithm details.
+//! 
+//! Current election algorithm is implemented based on Phragmén. The reference implementation can be found [here](https://github.com/w3f/consensus/tree/master/NPoS).
+//! 
+//! ## GenesisConfig
+//! 
+//! See the [`GensisConfig`] for a list of attributed that can be provided.
+//! 
+//! ## Related Modules 
+//! 
+//! - [**Balances**](https://crates.parity.io/srml_balances/index.html): Used to manage values at stake.
+//! - [**Sessions**](https://crates.parity.io/srml_session/index.html): Used to manage sessions. Also, a list of new validators is also stored in the sessions module's `Validators` at the end of each era.
+//! - [**System**](https://crates.parity.io/srml_system/index.html): Used to obtain block number and time, among other details.
+//! 
+//! # References
+//! 
+//! 1. This document is written as a more verbose version of the original [Staking.md](../Staking.md) file. Some sections, are taken directly from the aforementioned document.
```

Comment:
Didn't you remove this file in the last pr?

Reply:
Ugh, the intention was to maintain that as well. Though Gavin said he is also ok with the old one not being kept if everything is covered in a new doc. 

I will still keep this section and see where the missing file is + if it is recoverable.